tensorly/backend/paddle_backend.py

```diff
@@ -0,0 +1,366 @@
+                if current_device.startswith("gpu"):
+                    place = "cpu"
+
+            tensor = paddle.to_tensor(data, place=place)
+
+        # Set dtype/place/stop_gradient if specified
+        if dtype is not None:
+            tensor = tensor.astype(dtype)
+        if place is not None:
+            tensor = tensor.to(device=place)
+        if stop_gradient is not None:
+            tensor.stop_gradient = stop_gradient
+        return tensor
+
+    @staticmethod
+    def index_update(tensor: paddle.Tensor, indices, values):
+        """Updates the value of tensors in the specified indices
+            Should be used as::
+
+                index_update(tensor, tensorly.index[:, 3:5], values)
+
+            Equivalent of::
+
+                tensor[:, 3:5] = values
+
+        Parameters
+        ----------
+        tensor : tensorly.tensor
+            intput tensor which values to update
+        indices : tensorly.index
+            indices to update
+        values : tensorly.tensor
+            values to use to fill tensor[indices]
+
+        Returns
+        -------
+        tensor
+            updated tensor
+
+        Example
+        -------
+
+        >>> import tensorly as tl
+        >>> import numpy as np
+        >>> tensor = tl.tensor([[1, 2, 3], [4, 5, 6]])
+        >>> cpy = tensor.copy()
+        >>> tensor[:, 1] = 0
+        >>> tensor
+        array([[1, 0, 3],
+                [4, 0, 6]])
+        >>> tl.index_update(tensor, tl.index[:, 1], 0)
+        array([[1, 0, 3],
+               [4, 0, 6]])
+
+        See also
+        --------
+        index
+        """
+        if paddle.is_tensor(values):
+            if paddle.is_tensor(indices):
+                # indices is bool mask
+                mask_coord = paddle.concat(
+                    paddle.nonzero(indices, as_tuple=True),
+                    axis=1,
+                )
+                t = paddle.scatter_nd_add(
+                    tensor * (~indices).astype(tensor.dtype),
+                    mask_coord,
+                    values,
+                )
+                paddle.assign(t, tensor)  # inplace update
+            else:
+                # NOTE: Assure source Tensor is contiguous
+                tensor[indices] = (
+                    values if values.is_contiguous() else values.contiguous()
+                )
+        else:
+            tensor[indices] = values
+        return tensor
+
+    @staticmethod
+    def to_numpy(tensor: paddle.Tensor):
+        if paddle.is_tensor(tensor):
+            return tensor.numpy()
+        elif isinstance(tensor, np.ndarray):
+            return tensor
+        else:
+            return np.asarray(tensor)
+
+    @staticmethod
+    def shape(tensor: paddle.Tensor):
+        return tuple(tensor.shape)
+
+    @staticmethod
+    def ndim(tensor: paddle.Tensor):
+        return tensor.ndim
+
+    @staticmethod
+    def arange(start, stop=None, step=1.0, *args, **kwargs):
```

Comment:
Is this handled as a special case to avoid a typing error? From the documentation, it looks like you may be able to just import straight from paddle.

Reply:
Simplified to registering code